### ISSUE TITLE
Unable to evaluate variable

### ISSUE THREAD

**Issue by bhackett1024** [RecordReplay/devtools#1602]
STR

1. visit https://replay.io/view?id=d1b0ada5-7d39-447c-bd6e-5c5b2e073168
2. add a breakpoint at object-inspector/reducer.js:19
3. visit one of the breakpoint messages
4. evaluate "action" in console

Result: `ReferenceError: "action is not defined"`

Looking in the scopes panel, "action" has an entry, so we ought to be able to evaluate it as well.

**Comment by bhackett1024** [comment]
When looking at the generated source the breakpoint on line 19 actually ends up inside an inner function, and the variable used for `action` has been shadowed by another variable.  Actually getting the evaluation to work will be tricky: we need to be able to create a binding dynamically within the scope of the evaluation, but the firefox devtools support that though the CDP doesn't seem to.  An alternative would be to provide a better error message here.

**Comment by bhackett1024** [comment]
This isn't high priority to fix right now, the source map makes dealing with this difficult and the original scopes show what is needed to at this point.

**Comment by loganfsmyth** [comment]
The issue here is this, primarily, is that the cursor placement on this ends up coming up _inside_ the function, and because the code is minified as
```
function reducer(state = initialState(), action = {}) {
  const { type, data } = action;

  const cloneState = overrides => ({ ...state, ...overrides });
```
becomes
```
function foo(e, t = {}) {
    const {
      type: r,
      data: o
    } = t, i = t => ({
      ...e,
      ...t
    });
```
it actually isn't possible to reach the outer `t` from that scope because the `overrides` binding was renamed to `t` and shadows the outer `t`.

The right fix for this is probably to add some more logic about how we map a location to a particular scope, so that a cursor right at the beginning of a function expression doesn't count as _part_ of that function.

I'll come back to this but at least it seems to have enough requirements to reproduce that it wouldn't be super common.

The `reducer.js` file in the original recording doesn't exist anymore, but when I reproduced locally with a new test file (a66bdb60-fb62-46c2-a1d2-4a45b1a2c193), I also got different column breakpoint marker placement which confuses me.

From `d1b0ada5-7d39-447c-bd6e-5c5b2e073168`:
<img width="535" alt="Screen Shot 2021-05-13 at 1 10 46 PM" src="https://user-images.githubusercontent.com/132260/118181495-b0839800-b3ec-11eb-9ac3-7eba7a36e713.png">

vs locally:
<img width="645" alt="Screen Shot 2021-05-12 at 3 37 36 PM" src="https://user-images.githubusercontent.com/132260/118181620-cf822a00-b3ec-11eb-8e9f-5fe12b39549d.png">
